plugins/kubernetes-ingestor/src/auth/kubernetesCredentialProvider.ts

```diff
@@ -17,6 +17,15 @@ export async function getAuthCredential(
   config: Config,
   logger: LoggerService,
 ): Promise<KubernetesCredential> {
+  // First, check if we have a custom auth strategy registered globally
+  const globalAuthStrategies = (global as any).kubernetesAuthStrategies;
+  if (globalAuthStrategies && globalAuthStrategies.has(authProvider)) {
+    const strategy = globalAuthStrategies.get(authProvider);
+    logger.debug(`Using registered custom auth strategy: ${authProvider}`);
+    return await strategy.getCredential(cluster, {});
+  }
+
+  // Fallback to built-in auth providers
```

Comment:
i am not sure if we really need the fallback here when we use the globalAuthStrategies ?!